### PR TITLE
Centralize shell support utils

### PR DESCRIPTION
**Pull request by linsword13**
* Put a few shell utils into a common util file
* Add a `target_shells` directive for ramble objects, to declare explicit supported set of shells

Also a unrelated change to ensure license file checks don't complain on removed files.

### CODE REVIEW

File: lib/ramble/ramble/application.py [hunk 1]
```diff
@@ -72,6 +72,23 @@ def _get_context_display_name(context):
     )
 
 
+def _check_shell_support(app_inst):
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/application.py:75]: Should this also apply to modifiers and package managers?

Either in their base classes, or in the application class where the instances for the other objects are created?

File: lib/ramble/ramble/language/shared_language.py [hunk 2]
```diff
@@ -415,3 +415,21 @@ def _execute_tag(obj):
...
+    If not specified, i.e., not directly specified or inherited from the base,
+    then it assumes all shells are supported.
+
+    Args:
+        shell_support_pattern: The glob pattern that is used to match with the configured shell
+    """
+
+    def _execute_target_shells(obj):
+        if shell_support_pattern is not None:
+            obj.shell_support_pattern = shell_support_pattern
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/language/shared_language.py:433]: Odd question, but does this just create an instance attribute instead of a class attribute?

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]

File: lib/ramble/ramble/application.py (hunk 1 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/application.py:75]: Ah, never mind, sorry. I see that this handles checking the other objects too and that they aren't responsible for making sure they are valid

**Comment by douglasjacobsen** [comment]
Look good! There is a conflict, and I had one question.

File: lib/ramble/ramble/language/shared_language.py (hunk 2 above)
Comment (linsword13) [reply lib/ramble/ramble/language/shared_language.py:433]: Right now it's set as a class attribute, as it does not use the `dicts` argument to add into `_directive_names`. I think that's what I want as this shouldn't have per-instance modification, but maybe I miss something?

**Comment by linsword13** [comment]
> Look good! There is a conflict, and I had one question.

Thanks. Resolved the conflict. I will also hold off submitting this and wait for your info PR to get in. Then I will add support for this new attribute in this PR.

**Comment by linsword13** [comment]
> > Look good! There is a conflict, and I had one question.
> 
> Thanks. Resolved the conflict. I will also hold off submitting this and wait for your info PR to get in. Then I will add support for this new attribute in this PR.

Updated with the newest develop.

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**